src/libraries/System.Private.CoreLib/src/System/Guid.cs

```diff
@@ -197,6 +197,16 @@ internal GuidResult(GuidParseThrowStyle canThrow) : this()
             {
                 return Unsafe.As<GuidResult, Guid>(ref Unsafe.AsRef(in this));
             }
+
+            public void ReverseEndianness()
+            {
+                _a = BinaryPrimitives.ReverseEndianness(_a);
+                _b = BinaryPrimitives.ReverseEndianness(_b);
+                _c = BinaryPrimitives.ReverseEndianness(_c);
+                _de = BinaryPrimitives.ReverseEndianness(_de);
```

Comment:
I think only _a, _b, and _c need to be swapped here.   _de, _fg, and _hijk are already correct in either endianness.

As an aside, is there are (performance) reason to access the _a, _b, and _c fields via a byte-wise overlay?  If you'd just compute their values as 32-bit / 16-bit integers to begin with, it would be correct in either endianness and there wouldn't be any need for swapping at all.

Reply:
> I think only _a, _b, and _c need to be swapped here. _de, _fg, and _hijk are already correct in either endianness.

Fixed

> is there a (performance) reason to access the _a, _b, and _c fields via a byte-wise overlay? If you'd just compute their values as 32-bit / 16-bit integers to begin with, it would be correct in either endianness and there wouldn't be any need for swapping at all.

I'd done that initially, actually, but switched away mainly because it was more complicated code and didn't yield any measurable perf benefits.

Reply:
OK, this version passes libs.tests on s390x without failures.

Reply:
Excellent, thanks for checking.